File: app/src/main/java/com/yjq/knowledge/contract/ZhihuContract.java
package com.yjq.knowledge.contract;


import com.yjq.knowledge.beans.zhihu.ZhihuDaily;

import rx.Observable;

/**
 * 文件： ZhihuContract
 * 描述：
 * 作者： YangJunQuan   2017/12/15.
 */

public interface ZhihuContract {
    interface Ipresenter {
        void loadZhihuNews();

        void loadMoreZhihuNews(String data);
    }

    interface Iview {
        void showNews(ZhihuDaily zhihuDaily);

        void showMoreNews(ZhihuDaily zhihuDaily);

    }

    interface Imodel {

        Observable<ZhihuDaily> loadNews();

        Observable<ZhihuDaily> loadMore(String data);
    }
}

File: app/src/main/java/com/yjq/knowledge/contract/ZhihuContract.kt
package com.yjq.knowledge.contract


import com.yjq.knowledge.beans.zhihu.ZhihuDaily

import rx.Observable

/**
 * 文件： ZhihuContract
 * 描述：
 * 作者： YangJunQuan   2017/12/15.
 */

interface ZhihuContract {
    interface Ipresenter {
        fun loadZhihuNews()

        fun loadMoreZhihuNews(data: String)
    }

    interface Iview {
        fun showNews(zhihuDaily: ZhihuDaily)

        fun showMoreNews(zhihuDaily: ZhihuDaily)

    }

    interface Imodel {

        fun loadNews(): Observable<ZhihuDaily>

        fun loadMore(data: String): Observable<ZhihuDaily>
    }
}
